Failure to build with AOT disabled

<!--
If you are new to the project get yourself familiar with https://www.mono-project.com/community/bugs/make-a-good-bug-report/ before filling the issue
-->

## Steps to Reproduce

1. Seeing the below errors with attempting to build host-mono with configuration set to disable aot. 
This is using buildroot, from the configuration: 

> > MONO_COMMON_CONF_OPTS = --with-mcs-docs=no \
> 	--with-ikvm-native=no \
> 	--with-sigaltstack=no \
> 	--with-moonlight=no \
> 	--with-runtime_preset=net_4_x mono_cv_uscore=no \
> 	--enable-static \
> 	--disable-system-aot \
> 	--disable-gtk-doc \
> 	--enable-minimal=profiler,debug,aot \

<!--
You may drag & drop the attachment (repro code/solution, screenshot, etc.) onto the issue.
-->

### Current Behavior

> ./.libs/libmini.a(libmini_la-llvmonly-runtime.o): In function `init_llvmonly_method':
> /opt/buildroot/output/build/host-mono-custom/mono/mini/llvmonly-runtime.c:777: undefined reference to `mono_aot_init_llvmonly_method'
> collect2: ld returned 1 exit status
> make[6]: *** [mono-sgen] Error 1
> make[6]: *** Waiting for unfinished jobs....


## On which platforms did you notice this

[ ] macOS
[X ] Linux
[ ] Windows

**Version Used**:
This is building against the release version 6.4.0.198

No difference. 

> 
> make[6]: *** Waiting for unfinished jobs....
> ./.libs/libmono-ee-interp.a(libmono_ee_interp_la-interp.o): In function `memset':
> /usr/include/x86_64-linux-gnu/bits/string3.h:82: warning: memset used with constant zero length parameter; this could be due to transposed parameters
> ./.libs/libmini.a(libmini_la-llvmonly-runtime.o): In function `init_llvmonly_method':
> /opt/buildroot/output/build/host-mono-custom/mono/mini/llvmonly-runtime.c:777: undefined reference to `mono_aot_init_llvmonly_method'
>

--- a/src/mono/mono/mini/aot-compiler.c
+++ b/src/mono/mono/mini/aot-compiler.c
@@ -14263,4 +14263,11 @@ mono_aot_can_enter_interp (MonoMethod *method)
 	return FALSE;
 }
 
+int
+mono_aot_get_method_index (MonoMethod *method)
+{
+	g_assert_not_reached ();
+	return 0;
+}
+
 #endif

--- a/src/mono/mono/mini/aot-runtime.c
+++ b/src/mono/mono/mini/aot-runtime.c
@@ -6466,4 +6466,11 @@ mono_aot_get_method_flags (guint8 *code)
 	return MONO_AOT_METHOD_FLAG_NONE;
 }
 
+gboolean
+mono_aot_init_llvmonly_method (gpointer aot_module, guint32 method_index, MonoClass *init_class, MonoError *error)
+{
+	g_assert_not_reached ();
+	return FALSE;
+}
+
 #endif

